libvast/vast/view.hpp

```diff
@@ -178,6 +172,45 @@ struct container_view;
 template <class T>
 using container_view_ptr = caf::intrusive_ptr<container_view<T>>;
 
+/// @relates container_view
+template <class Pointer>
+class container_view_handle
+  : detail::totally_ordered<container_view_handle<Pointer>> {
+public:
+  container_view_handle() = default;
+
+  container_view_handle(Pointer ptr) : ptr_{ptr} {
+    // nop
+  }
+
+  explicit operator bool() const {
+    return static_cast<bool>(ptr_);
+  }
+
+  auto operator->() const {
+    return ptr_.get();
+  }
+
+  const auto& operator*() const {
+    return *ptr_;
+  }
+
+private:
+  Pointer ptr_;
+};
+
+template <class Pointer>
+bool operator==(const container_view_handle<Pointer>& x,
+                const container_view_handle<Pointer>& y) {
+  return x && y && *x == *y;
+}
+
+template <class Pointer>
+bool operator<(const container_view_handle<Pointer>& x,
+               const container_view_handle<Pointer>& y) {
+  return x && y && *x < *y;
```

Comment:
Doesn't this implementation break total ordering, i.e., `(x < nullptr) == (nullptr < x)`?